tools/integration-test/src/tests/multihop/handshake.rs

```diff
@@ -0,0 +1,211 @@
+use ibc_relayer::channel::{extract_channel_id, Channel, ChannelSide};
+use ibc_relayer::registry::set_global_registry;
+use ibc_relayer_types::core::ics33_multihop::channel_path::ConnectionHops;
+use ibc_test_framework::prelude::*;
+use ibc_test_framework::relayer::channel::assert_eventually_multihop_channel_established;
+use ibc_test_framework::relayer::connection::create_connection_hop;
+
+#[test]
+fn test_multihop_channel_open_handshake() -> Result<(), Error> {
+    run_nary_connection_test(&MultihopChannelHandshakeTest)
+}
+
+pub struct MultihopChannelHandshakeTest;
+
+impl TestOverrides for MultihopChannelHandshakeTest {
+    fn modify_test_config(&self, config: &mut TestConfig) {
+        config.bootstrap_with_random_ids = false;
+    }
+
+    fn modify_relayer_config(&self, config: &mut Config) {
+        config.mode.clients.misbehaviour = false;
+    }
+
+    fn should_spawn_supervisor(&self) -> bool {
+        false
+    }
+}
+
+/// FIXME: Currently in order to have the correct channel at each step
+/// the Channel is built manually.
+/// Once the restore_from_event has been updated to be compatible with
+/// multihop this test can be improved by using restore_from_event at each
+/// step.
+impl NaryConnectionTest<3> for MultihopChannelHandshakeTest {
+    fn run<Handle: ChainHandle>(
+        &self,
+        _config: &TestConfig,
+        relayer: RelayerDriver,
+        chains: NaryConnectedChains<Handle, 3>,
+        connections: ConnectedConnections<Handle, 3>,
+    ) -> Result<(), Error> {
+        let registry = SharedRegistry::new(relayer.config.clone());
+        set_global_registry(registry.clone());
+        let chain_handle_a = chains.chain_handle_at::<0>()?;
+        let chain_handle_b = chains.chain_handle_at::<1>()?;
+        let chain_handle_c = chains.chain_handle_at::<2>()?;
+
+        let client_a_to_b = chains.foreign_client_at::<1, 0>()?;
```

Comment:
A nitpick, but I think it would be clearer to have the client that tracks chain B on chain A be named `client_b_on_a`. But if there's a convention already let's just keep it this way.